src/Qir/Runtime/public/QubitManager.hpp

```diff
@@ -185,6 +183,7 @@ namespace Quantum
         public:
             QubitListInSharedArray FreeQubitsReuseProhibited;
             QubitListInSharedArray FreeQubitsReuseAllowed;
+            int32_t prevAreaWithFreeQubits = 0;
```

Comment:
I would add a comment explaining this logic. At lease mention that this is a hint. This previous area with free qubits might not actually have free qubits. It is more of a "skip to" area. it is guaranteed that there are no free qubits between current area and this skip-to area.

Reply:
Correct. The guarantee is that there're no free qubits between current area and the area pointed to by this variable (bounds non-inclusive). However it would be non-practical to name the variable to reflect exact definition. So I think I'll keep the variable name. I'll definitely add the comments.